viadot/sources/bigquery.py

```diff
@@ -0,0 +1,113 @@
+from typing import List
+import pandas_gbq
+import pandas as pd
+from google.oauth2 import service_account
+
+from ..config import local_config
+from ..exceptions import CredentialError, DBDataAccessError
+from .base import Source
+
+
+class BigQuery(Source):
+    """
+    Class to connect with Bigquery project and tables.
+
+    Note that credentials used for authentication can be generated only for User Principal
+    who has access to specific BigQuery project.
+    """
+
+    def __init__(self, credentials_key: str = None, *args, **kwargs):
+        """
+        Create an instance of BigQuery class.
+
+        Args:
+            credentials_key (str): Credential key to dictionary where details are stored. Credentials can be generated as key
+            for User Principal inside a BigQuery project. Defaults to None.
+
+        Raises:
+            CredentialError: In case credentials cannot be found.
+        """
+        self.credentials_raw = local_config.get(credentials_key)
+        if self.credentials_raw is None:
+            raise CredentialError("Credentials not found.")
+
+        super().__init__(*args, credentials=self.credentials_raw, **kwargs)
+
+        credentials = service_account.Credentials.from_service_account_info(
```

Comment:
We should add the option for Azure KV storage - it's not mandatory but it should be added in the next version or even now.